TicTacToe/TicTacToe.cs

```diff
@@ -32,46 +37,114 @@ public static void GetInput()
             int row = int.Parse(Console.ReadLine());
             Console.WriteLine("Enter Column:");
             int column = int.Parse(Console.ReadLine());
+            PlaceMark(row, column);
+            CheckForWin();
+            CheckForTie();
+            playerTurn = (playerTurn == "X") ? "O" : "X";
+
         }
 
         public static void PlaceMark(int row, int column)
         {
-        // your code goes here
+
+            //string[] myRow = board[row];
+            //string myColumn = myRow[column];
+            //myRow[column] = "X";
+            board[row][column] = playerTurn;
+
+
+
+
+
+
+            // your code goes here
         }
 
         public static bool CheckForWin()
-        {
-            // your code goes here
+        { // your code goes here
 
+            if (HorizontalWin() || VerticalWin() || DiagonalWin() == true)
+            {
+                Console.WriteLine("Player " + playerTurn + " Won!");
+                Environment.Exit(0);
```

Comment:
No need to call `Environment.Exit` here, just return true and your while loop should exit